### PR TITLE
Bug 1711043: ignore mgmt state of Removed

### PR DESCRIPTION
**Pull request by jboyd01**
for 4.1 GA, ignore any change to managementStage because when we were removing the namespace the following would happen:
1) We delete the namespace but we don't delete the Service Catalog API Service
2) because of the aggregated api service still existing but now not available, namespaces will get stuck in a terminating state
3) if we delete the api service, GC will delete all secrets associated with service catalog bindings (because of the ownerRef), this may result in accidental data loss.

For 4.1, add a condition indicating we are ignoring the desired state to removed and consult the doc.  Doc will give steps to remove owner ref from secrets (hopefully simple oc script) and then tell uses to delete the openshift-service-catalog-apiserver namespace.  This will result in removal of service catalog api server (the FinalizerController sync() will kick in).

condition looks like this:
```
status:
  conditions:
....
  - lastTransitionTime: 2019-05-17T00:19:05Z
    message: Unable to automatically remove Service Catalog once installed. Request
      to remove will be ignored. See documentation for more details.
    reason: Unable to automatically remove Service Catalog once installed.
    status: "True"
    type: RemovalRequestIgnored
```

if cluster admin then deletes the namespace the operator cleans out the deployment and we are back to a happy state of Removed and the condition is cleared.  Condition is also cleared if admin toggles from initial Removed to Managed or Unmanabed.

Need the following review:
@jwforres 
@shawn-hurley
@jmrodri 
@bparees additional review wouldn't hurt if you have time.

I have tested various scenarios taking this from Removed -> Managed -> Removed -> Managed -> Removed -> manually delete namespace  -> observe catalog is fully removed -> Managed -> observe catalog is operational

The new condition comes and goes as it should.  It lgtm.

I'm adding a hold, @jwforres  please remove when you are satisfied.
/hold

### CODE REVIEW

**Review by shawn-hurley — APPROVED** [review]
lgtm

**Comment by jwforres** [comment]
lgtm as well and has necessary approvals

/hold cancel

File: pkg/operator/workloadcontroller/workload_controller.go
Comment (bparees) [inline pkg/operator/workloadcontroller/workload_controller.go:149]: sorry i'm getting here late, but this is a little loose as a check...  you could get an err on this Get() for any number of reasons (e.g. network or api server hiccup) and you're defaulting to assuming it's ok to proceed w/ the removal if you get any error.  Ideally i think you should be confirming this err is a NotFound error before assuming it's safe to proceed with a normal removal.

Comment (jwforres) [reply pkg/operator/workloadcontroller/workload_controller.go:149]: We don't actually do any removal now, so it will just mean we don't report the "removal ignored" status. So there is no serious risk, just imperfect failure reporting.

Comment (bparees) [reply pkg/operator/workloadcontroller/workload_controller.go:149]: i see now, thanks

**Pull request merged**
